src/__utils/Overlay/Overlay.tsx

```diff
@@ -59,22 +59,22 @@ const Overlay: React.FC<OverlayProps> = React.forwardRef((props: OverlayProps, r
      * onScroll handler
      * @param {Event} ev The window scroll event
      */
-    function onScroll(ev: Event): void {
+    async function onScroll(ev: Event) {
         const target: HTMLElement = ev.target as HTMLElement;
         if (props.show && target.contains(props.overlayReference())) {
             const referenceDomRect: DOMRect = props.overlayReference().getBoundingClientRect();
             if (referenceDomRect.bottom < 0 || referenceDomRect.right < 0 || referenceDomRect.left > window.innerWidth || referenceDomRect.top > window.innerHeight) {
                 overlayContentRef?.current?.blur();
             }
-            setPlacementWithCoords(overlayPositionChecker.getPosition(props.position || "top"));
+            setPlacementWithCoords(await overlayPositionChecker.getPosition(props.position || "top"));
```

Comment:
`OverlayPositionChecker.getPosition()` is not asynchronous and neither are any of the methods that are being called inside so if adding async before this makes it work it probably means the underlying issue lies somewhere else in your code. You're probably calling something before something else or not waiting for something. I suspect adding a setTimeout instead of async/await would also work but again that is also a kind of hack.

Reply:
Of course I could be wrong and the underlying DOM manipulation APIs are in fact asynchronous and are supposed to be used like this in which case I would love to see some documentation or references to practices which do this.

Reply:
i think i should convert `getPosition` to async as well

Reply:
You should convert getPosition to async if whatever you're doing inside getPosition is in fact asynchronous. That's the main question.